### PR TITLE
bindinfo: migrate test-infra to testify(staging-2)

### LINKED ISSUE #27177: migrate test-infra to testify for `bindinfo` pkg

### PR DESCRIPTION
**Pull request by w169q169**
### What problem does this PR solve?

Issue Number: ref #27177 

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
@w169q169 I guess this one cannot "close" the issue, so you can use "to" or "ref" which has no magic but a reference.

File: bindinfo/session_handle_test.go [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+	plannercore "github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/stmtsummary"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGlobalAndSessionBindingBothExist(t *testing.T) {
+	store, _, clean := testkit.CreateMockStoreAndDomain(t)
```
Comment (tisonkun) [inline bindinfo/session_handle_test.go:36]:
```suggestion
	store, clean := testkit.CreateMockStore(t)
```

If you can want the `domain`, you can use this method. ditto if others.

File: bindinfo/session_handle_test.go [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+	"github.com/pingcap/tidb/metrics"
+	plannercore "github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/stmtsummary"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGlobalAndSessionBindingBothExist(t *testing.T) {
```
Comment (tisonkun) [inline bindinfo/session_handle_test.go:35]: If these tests are run in serial, please name the file "xxx_serial_test.go"

File: bindinfo/session_handle_test.go [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+	tk.MustExec("drop database if exists SPM")
+	tk.MustExec("create database SPM")
+	tk.MustExec("use SPM")
+	tk.MustExec("create table t(a int, b int)")
+	require.True(t, tk.Session().Auth(&auth.UserIdentity{Username: "root", Hostname: "%"}, nil, nil))
+	tk.MustExec("update t set a = a + 1")
+	tk.MustExec("update t set a = a + 1")
+	tk.MustExec("admin capture bindings")
+	rows := tk.MustQuery("show global bindings").Rows()
+	require.Equal(t, 1, len(rows))
```
Comment (tisonkun) [inline bindinfo/session_handle_test.go:189]:
```suggestion
	require.Len(t, rows, 1)
```

ditto others

File: bindinfo/session_handle_test.go [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+
+		metrics.BindTotalGauge.Reset()
+		metrics.BindMemoryUsage.Reset()
+
+		_, err := tk.Exec("create session " + testSQL.createSQL)
+		require.Nil(t, err, "err %v", err)
+
+		if testSQL.overlaySQL != "" {
+			_, err = tk.Exec("create session " + testSQL.overlaySQL)
+			require.Nil(t, err)
```
Comment (tisonkun) [inline bindinfo/session_handle_test.go:110]:
```suggestion
			require.NoError(t, err)
```

ditto others

File: bindinfo/session_handle_test.go (hunk 2 above)
Comment (tisonkun) [reply bindinfo/session_handle_test.go]: @w169q169 "these tests" mean all of the tests in this file, not only the commented one.

The rationale is that we'd prefer to run test in parallel for efficiency, and in order to help contributors be aware of whether a bunch of tests run in parallel or not, make serial test secondary and move then all in "xxx_serial_test.go"

Comment (w169q169) [reply bindinfo/session_handle_test.go]: Okay, I got it. I make it wrong in the previous commit and I have fixed it.

**Review by tisonkun — APPROVED** [review]

**Comment by w169q169** [comment]
/cc @morgo 
PTAL

File: bindinfo/session_handle_serial_test.go [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int, index idx(a))")
+	tk.MustExec("create global binding for select * from t using select * from t use index(idx);")
+	rows := tk.MustQuery("show global bindings").Rows()
+	require.Len(t, rows, 1)
+	createTime := rows[0][4]
+	time.Sleep(1000000)
```
Comment (morgo) [inline bindinfo/session_handle_serial_test.go:325]: I know this is copied from the original, but can se change it to a const for readability:
```suggestion
	time.Sleep(time.Millisecond)
```

File: bindinfo/session_handle_serial_test.go [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+	tk.MustExec("create global binding for select * from t using select * from t use index(idx);")
+	rows = tk.MustQuery("show global bindings").Rows()
+	require.Len(t, rows, 1)
+	require.Equal(t, false, createTime == rows[0][4])
+
+	tk.MustExec("create session binding for select * from t using select * from t use index(idx);")
+	rows = tk.MustQuery("show session bindings").Rows()
+	require.Len(t, rows, 1)
+	createTime = rows[0][4]
+	time.Sleep(1000000)
```
Comment (morgo) [inline bindinfo/session_handle_serial_test.go:335]: Same as above: 
```suggestion
	time.Sleep(time.Millisecond)
```

File: bindinfo/session_handle_serial_test.go [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+	tk.MustExec("create table t(a int, b int, index idx(a))")
+	tk.MustExec("create global binding for select * from t using select * from t use index(idx);")
+	rows := tk.MustQuery("show global bindings").Rows()
+	require.Len(t, rows, 1)
+	createTime := rows[0][4]
+	time.Sleep(1000000)
+	tk.MustExec("create global binding for select * from t using select * from t use index(idx);")
+	rows = tk.MustQuery("show global bindings").Rows()
+	require.Len(t, rows, 1)
+	require.Equal(t, false, createTime == rows[0][4])
```
Comment (morgo) [inline bindinfo/session_handle_serial_test.go:329]: It seems like there is a better way to write this? Like [require.NotEqual](https://pkg.go.dev/github.com/stretchr/testify/require#NotEqual).

**Review by morgo — CHANGES_REQUESTED** [review]
Minor nits, otherwise LGTM

**Review by morgo — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your review @morgo ! Reapply my approval.

**Pull request merged**